src/TelemetryConsumers/Orleans.TelemetryConsumers.Linux/LinuxEnvironmentStatistics.cs

```diff
@@ -0,0 +1,276 @@
+
+                var currentCpuUsage = (1.0f - deltaIdleTime / ((float)deltaTotalTime)) * 100f;
+
+                var previousCpuUsage = CpuUsage ?? 0f;
+                CpuUsage = (previousCpuUsage + 2 * currentCpuUsage) / 3;
+            }
+
+            _prevIdleTime = idleTime;
+            _prevTotalTime = totalTime;
+        }
+
+        private async Task UpdateAvailableMemory()
+        {
+            var memAvailableLine = await ReadLineStartingWithAsync("/proc/meminfo", "MemAvailable");
+
+            if (string.IsNullOrWhiteSpace(memAvailableLine))
+            {
+                _logger.LogWarning("Couldn't read 'MemAvailable' line from '/proc/meminfo'");
+                return;
+            }
+
+            if (!long.TryParse(new string(memAvailableLine.Where(char.IsDigit).ToArray()), out var availableMemInKb))
+            {
+                _logger.LogWarning($"Couldn't parse meminfo output: '{memAvailableLine}'");
+                return;
+            }
+
+            AvailableMemory = availableMemInKb * 1_000;
+        }
+
+        private void WriteToStatistics()
+        {
+            FloatValueStatistic.FindOrCreate(StatisticNames.RUNTIME_CPUUSAGE, () => CpuUsage.Value);
+            IntValueStatistic.FindOrCreate(StatisticNames.RUNTIME_GC_TOTALMEMORYKB, () => (long)((MemoryUsage + KB - 1.0) / KB)); // Round up
+
+#if LOG_MEMORY_PERF_COUNTERS
+            IntValueStatistic.FindOrCreate(StatisticNames.RUNTIME_MEMORY_TOTALPHYSICALMEMORYMB, () => (long)((TotalPhysicalMemory / KB) / KB));
+            IntValueStatistic.FindOrCreate(StatisticNames.RUNTIME_MEMORY_AVAILABLEMEMORYMB, () => (long)((AvailableMemory / KB) / KB)); // Round up
+#endif
+
+            IntValueStatistic.FindOrCreate(StatisticNames.RUNTIME_DOT_NET_THREADPOOL_INUSE_WORKERTHREADS, () =>
+            {
+                ThreadPool.GetMaxThreads(out var maXworkerThreads, out var maXcompletionPortThreads);
+
+                // GetAvailableThreads Retrieves the difference between the maximum number of thread pool threads
+                // and the number currently active.
+                // So max-Available is the actual number in use. If it goes beyond min, it means we are stressing the thread pool.
+                ThreadPool.GetAvailableThreads(out var workerThreads, out var completionPortThreads);
+                return maXworkerThreads - workerThreads;
+            });
+            IntValueStatistic.FindOrCreate(StatisticNames.RUNTIME_DOT_NET_THREADPOOL_INUSE_COMPLETIONPORTTHREADS, () =>
+            {
+                ThreadPool.GetMaxThreads(out var maxWorkerThreads, out var maxCompletionPortThreads);
+
+                ThreadPool.GetAvailableThreads(out var workerThreads, out var completionPortThreads);
+                return maxCompletionPortThreads - completionPortThreads;
+            });
+        }
+
+        private async Task Monitor(CancellationToken ct)
+        {
+            for (int i = 0; ; i++)
```

Comment:
Not a big fan of letting a int getting incremented "forever". Can you reset this counter in this loop at least?